genvm/core/context.go

```diff
@@ -38,16 +39,40 @@ type Context struct {
 }
 
 // Spawn account.
-func (c *Context) Spawn(template Address, args scale.Encodable) error {
-	principal := ComputePrincipal(template, args)
-	// TODO(dshulyak) only self-spawn is supported
-	if principal != c.Header.Principal {
-		return ErrSpawn
+func (c *Context) Spawn(args scale.Encodable) error {
+	template := c.Header.Template
+	address := ComputePrincipal(template, args)
```

Comment:
can you rename address/account to spawned/spawnedAcct for easier differentiation with c.Account.Address